src/protonpack/java/com/codepoetics/protonpack/AggregatingSpliterator.java

```diff
@@ -15,8 +17,8 @@
     private List<I> currentSlide = new ArrayList<>();
 
     AggregatingSpliterator(Spliterator<I> source, BiPredicate<List<I>, I> predicate) {
-        this.source = source;
-        this.condition = predicate;
+        this.source = requireNonNull(source);
+        this.condition = requireNonNull(predicate);
```

Comment:
Hmm ok, we can still move it into the factory method but if they are many that could lead to code duplication... Well feel free to reject this PR :-)